tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -3392,3 +3393,305 @@ def call(self, inputs, state):
+
+    self.num_parameters_per_head = (
+      self.memory_vector_dim + 2 * self.shift_range + 4)
+    self.num_heads = self.read_head_num + self.write_head_num
+    self.total_parameter_num = (
+      self.num_parameters_per_head * self.num_heads +
+      self.memory_vector_dim * 2 * self.write_head_num)
+
+  @property
+  def state_size(self):
+    return NTMControllerState(
+      controller_state=self.controller.state_size,
+      read_vector_list=[self.memory_vector_dim
+        for _ in range(self.read_head_num)],
+      w_list=[self.memory_size
+        for _ in range(self.read_head_num + self.write_head_num)],
+      M=tensor_shape.TensorShape([self.memory_size * self.memory_vector_dim]),
+      time=tensor_shape.TensorShape([]))
+
+  @property
+  def output_size(self):
+    return self.output_dim
+
+  def build(self, inputs_shape):
+    if self.output_dim is None:
+      if inputs_shape[1].value is None:
+        raise ValueError("Expected inputs.shape[-1] to be known, saw shape: %s"
+                         % inputs_shape)
+      else:
+        self.output_dim = inputs_shape[1].value
+
+    def create_linear_initializer(input_size, dtype=dtypes.float32):
+      stddev = 1.0 / math.sqrt(input_size)
+      return init_ops.truncated_normal_initializer(stddev=stddev, dtype=dtype)
+
+    self._params_kernel = self.add_variable(
+        'parameters_kernel',
+        shape=[self.controller.output_size, self.total_parameter_num],
+        initializer=create_linear_initializer(self.controller.output_size))
+
+    self._params_bias = self.add_variable(
+        'parameters_bias',
+        shape=[self.total_parameter_num],
+        initializer=init_ops.constant_initializer(0.0, dtype=self.dtype))
+
+    self._output_kernel = self.add_variable(
+        'output_kernel',
+        shape=[self.controller.output_size +
+          self.memory_vector_dim * self.read_head_num, self.output_dim],
+        initializer=create_linear_initializer(
+          self.controller.output_size +
+          self.memory_vector_dim * self.read_head_num))
+
+    self._output_bias = self.add_variable(
+        'output_bias',
+        shape=[self.output_dim],
+        initializer=init_ops.constant_initializer(0.0, dtype=self.dtype))
+
+    self._init_read_vectors = [self.add_variable(
+        'initial_read_vector_%d' % i,
+        shape=[1, self.memory_vector_dim],
+        initializer=initializers.xavier_initializer())
+        for i in range(self.read_head_num)]
+
+    self._init_address_weights = [self.add_variable(
+        'initial_address_weights_%d' % i,
+        shape=[1, self.memory_size],
+        initializer=initializers.xavier_initializer())
+        for i in range(self.read_head_num + self.write_head_num)]
+
+    self.built = True
+
+  def call(self, x, prev_state):
+    # Addressing Mechanisms (Sec 3.3)
+
+    prev_read_vector_list = control_flow_ops.cond(
+      math_ops.equal(prev_state.time, 0),
+      lambda: [self._expand(
```

Comment:
long lambdas are against our style guide.  can you move this to a def above, lke:

def _prev_read_vector_list_initial_value():
  ...

then here, use cond(time == 0, _prev_read_vector_list_initial_value, ...)